components/availability/src/Availability.svelte

```diff
@@ -346,6 +345,43 @@
     });
   //#endregion layout
 
+  // #region timeSlot selection
+  let slotSelection: SelectableSlot[] = [];
+  let sortedSlots: SelectableSlot[] = [];
+  $: slotSelection = days
+    .map((day) =>
+      day.slots.filter(
+        (slot) => slot.selectionStatus === SelectionStatus.SELECTED,
+      ),
+    )
+    .flat();
+  $: sortedSlots = slotSelection
+    .sort((a, b) => a.start_time.getTime() - b.start_time.getTime())
+    .reduce((slotList, slot) => {
+      const prevSlot = slotList[slotList.length - 1];
+      if (
+        prevSlot &&
+        slot.start_time.getTime() ===
+          prevSlot[prevSlot.length - 1].end_time.getTime()
+      ) {
+        prevSlot.push(slot);
+      } else {
+        slotList.push([slot]);
+      }
+      return slotList;
+    }, [] as SelectableSlot[][])
+    .map((slotList) => {
+      return {
+        ...slotList[0],
+        end_time: slotList[slotList.length - 1].end_time,
+      };
+    });
+
+  $: if (sortedSlots.length) {
```

Comment:
- Need to handle the case where slots were previously selected, but now all have been deselected
- Should consider doing a diff with previous sortedSlots